packages/core/src/connectors/facebook/index.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Reference: Manually Build a Login Flow
+ * https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow
+ */
+import got, { RequestError as GotRequestError } from 'got';
+import { stringify } from 'query-string';
+import { z } from 'zod';
+
+import {
+  accessTokenEndpoint,
+  authorizationEndpoint,
+  scope,
+  userInfoEndpoint,
+} from '@/connectors/facebook/constant';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  ConnectorType,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ValidateConfig,
+} from '@/connectors/types';
+import { getConnectorConfig, getConnectorRequestTimeout } from '@/connectors/utilities';
+
+export const metadata: ConnectorMetadata = {
+  id: 'facebook',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+};
+
+const facebookConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+});
+
+type FacebookConfig = z.infer<typeof facebookConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  if (!config) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, 'Missing config');
+  }
```

Comment:
these lines of code are unnecessary and are removed in `aliyun-dm` and `google connector`, for the code below ensure this configuration should not be empty.

Reply:
Did not follow here.  
As `"these lines of code are unnecessary and are removed in aliyun-dm and google connector"` indicates the non-empty assertion is unnecessary.  
But `"for the code below ensure this configuration should not be empty" `says it is needed.  🤣

Reply:
Get it. Since `githubConfigGuard.safeParse` can assert it non-empty, removed it. PTAL.

Reply:
Related PR #329